### ISSUE TITLE
[BUG]: tools do not have binary path hardcoded in its arguments

### ISSUE THREAD

**Issue by disconnect3d** [compiler-explorer/compiler-explorer#4773]
### Describe the bug

When adding a tool output such as readelf, ldd, pahole or strings, the tool itself needs a specific flag with the executable/binary file path. However, this is not provided by default and it is uncertain what path to provide.

### Steps to reproduce

1. Visit godbolt.org
2. Choose C example/compiler
3. Click "Add tool..." -> "readelf" or other tools

The tool output says "Strings requires a binary output" or "readelf requires an executable".

### Expected behavior

I expect the added tools to have added the argument to provide path to the resulting binary.

### Reproduction link

https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1AB9U8lJL6yAngGVG6AMKpaAVxYM9DgDJ4GmADl3ACNMYhAAVlIAB1QFQlsGZzcPPVj4mwFffyCWUPCoy0xrRKECJmICZPdPLgtMK0yGMoqCbMCQsMiLcsrq1LqFXra/DryuiIBKC1RXYmR2DgB6ACoAagAVAE9ozDWt2eI1tCw1hDDMUjWSNdpUJnQ1wzXMVVZo%2BgA6NZWlgFINABBPwENYKACOrgqmAgILWDHckzWfwA7AAhAGAtbYtbETAEOYMeHuH7Elh/ADMGKBqIAIpiONNaJwIrxPBwtKRUJxHGDDgtkQAmCk8UgETSM6YAaxAkgAbJ8AJwADgiyq4ETlGkFyopcuVyv0nEkbIlXM4vAUIA0Yol0zgsBgiBQqBY0ToYXIlDQbo94WQwC4gsFpCwADc8AsAGp4TAAdwA8rt2aKaLQCGErRBgmbgn4KltOKK88xiFsE8FtMVxdxeD62IIEwxaIWObwsCxDMBxG3Q3g8SUw5grb3XsVXBmzSCGmbaHhgsQC84sGaCMQ8Cwi4y%2BAZgAoY/Gk4wtzJBCIxOwpKf5Eo1GbdHUDEYQKZjOY58ErZBpqhok0R5aDTVok9gME4Lg1F4YHtLk%2BRpHECQCP0tQxAhTQwZ04SDEBJQCC0fQQQM9SNKUwwYeMWE9K0yF6EMrTkXBXDTAo/KXkyLKmr23IcGsqjKnKAC0cqSGswDIMgaxBp8gprBAuCEDcfzCkxvA1lokzSiAOqfBoUhcMqkgUoqKJcHKIqCtIzIcCapCboKUTspy3GWtatptvaToQEgswENEE5ehAPruvQxABKwix8YJwmieJkmCtJvCYPgRDrugej8GeojiFeGU3io6i9g%2BpBxou0RbuxHCsqQjm8NxCYTn5oKoFQvH8UJIliRJUkyRAzi%2BiFQoipMql2tM5wPF0EAVTZdkOWazkWK5anbg6zpEC4AUVHuFqkFtCjKIYDRCAgqBximdausFYRhWwnCRe1MVdfFgqJclJB4GldS5ee2XSLlij5feegsAITBoPJKUABJMAoaKYIwgITqgR7%2BGlta2aD4NvcQKMniDzBoGU%2BDnT0xM7UMxMHf4tDHadJPrbQACSaVDOuRgKMVpXlUalWcU5nCs34e6tVFHWxd1skQ4pylXH1V1HEpQ0je5028HZkhKoqWvazr2tytV83k0tdqkKtXkgAzm3ENtHC8HtVNHSdZ0nkFfo3RFbXRZ1cUJaQSUKal6WyD9l5/bIAN3oVwOY6gUvENDsPwwwiNECjSV4zHce4%2Bj%2BNg6gRN%2BCeFOF7bpN%2BA7NNO/TqAuMzIC0Og6Cc0wZW1hVVU1eaHAN4891e%2BLz2S9jg11Gsct%2BiPw1uepqu2Vpc1cUbNrLRppAyhEFI6XpBlGSZZlBjzFJ87VO0rxVL0G4vpdn6QQ7EPEdiSEAA

### Screenshots

<img width="328" alt="image" src="https://user-images.githubusercontent.com/10009354/221177209-3c35aacd-3456-43d4-8bc1-7612041924f8.png">

**Comment by disconnect3d** [comment]
Yeah, adding "Link to binary" works. It would be nice if this is checkboxed/enabled automatically when I add the tool output OR if the tool output said to do so if it is not enabled.

**Comment by partouf** [comment]
Not a fan of automatically enabling it, but better error messages would be great.

**Comment by jeremy-rifkin** [comment]
Maybe resolved by #5057
